go.mod

```diff
@@ -1,190 +1,330 @@
 module github.com/gardener/test-infra
 
-go 1.20
+go 1.21
 
 require (
-	cloud.google.com/go/storage v1.32.0
-	github.com/Masterminds/semver/v3 v3.2.0
+	cloud.google.com/go/storage v1.35.1
+	github.com/Masterminds/semver/v3 v3.2.1
 	github.com/Masterminds/sprig v2.22.0+incompatible
 	github.com/Masterminds/sprig/v3 v3.2.3
-	github.com/argoproj/argo-workflows/v3 v3.4.13
-	github.com/bradleyfalzon/ghinstallation/v2 v2.6.0
-	github.com/fsnotify/fsnotify v1.6.0
-	github.com/gardener/component-cli v0.55.0
-	github.com/gardener/component-spec/bindings-go v0.0.98
-	github.com/gardener/gardener v1.78.0
-	github.com/ghodss/yaml v1.0.1-0.20190212211648-25d852aebe32
+	github.com/argoproj/argo-workflows/v3 v3.5.1
```

Comment:
Please keep the argo dependency at its original version v3.4.13. Upgrading it should be done in a separate PR incl the deployed version of the argo controller. Since it is a minor patch, it will probably include some more changes.